pkg/sinks/sink.go

```diff
@@ -187,7 +219,22 @@ func (u *SinkProcessor) Start(ctx context.Context) error {
 			}
 		}
 
-		// NOTE: forwarder can make multiple sink writers when we introduce fallback sinks
+		// create the main sink writer
+		sinkWriter, err := u.createSinkWriter(ctx, &u.VertexInstance.Vertex.Spec.Sink.AbstractSink, sinkHandler)
+		if err != nil {
+			return fmt.Errorf("failed to find a sink, error: %w", err)
+		}
+
+		// create the fallback sink writer if fallback sink is present
+		if u.VertexInstance.Vertex.Spec.Sink.Fallback != nil {
```

Comment:
This requires a sink must be defined in the field `fallback`. We could argue this is a requirement, but usually this should not be enforced - if nothing is defined under `fallback`, we should consider there's no fallback specified.